packages/hooks/src/createHasHook.js

```diff
@@ -5,17 +5,25 @@
  * @param  {Object}   hooks Stored hooks, keyed by hook name.
  *
  * @return {Function}       Function that returns whether any handlers are
- *                          attached to a particular hook.
+ *                          attached to a particular hook and optional namespace.
  */
 function createHasHook( hooks ) {
 	/**
-	 * Returns how many handlers are attached for the given hook.
+	 * Returns whether any handlers are attached for the given hookName and optional namespace.
 	 *
-	 * @param  {string}  hookName The name of the hook to check for.
+	 * @param {string} hookName  The name of the hook to check for.
+	 * @param {string} namespace Optional. The unique namespace identifying the callback in the form `vendor/plugin/function`.
 	 *
 	 * @return {boolean} Whether there are handlers that are attached to the given hook.
 	 */
-	return function hasHook( hookName ) {
+	return function hasHook( hookName, namespace ) {
+		// Use the namespace if provided.
+		if ( 'undefined' !== typeof namespace ){
+			return hookName in hooks &&
+				hooks[ hookName ].handlers &&
```

Comment:
When will this not be truthy?

Reply:
>generally prefer the defensive approach of always checking before using because otherwise everything breaks.

Well, there's defensive, but there's also having a solid understanding of the shapes of data we expect to occur.

Reply:
True :)  I'll take another look - if handlers is always going to be truthy, i will remove the check.

Reply:
removed extraneous check